agent/recognition/binarymatch.py

```
@@ -614,28 +632,50 @@ def _run_preset(self, context: Context,
+
+        if reco.hit:
+            bx, by, bw, bh = reco.box
             adjusted = (bx + rx, by + ry, bw, bh)
-            print(f"[RedDotDetector] [preset:{preset_node}] 命中 → {adjusted}")
-            return CustomRecognition.AnalyzeResult(box=adjusted, detail={"preset": preset_node})
-        return None
+            mfaalog.info(f"[RedDotDetector] [preset:{preset_node}] hit -> {adjusted}")
+            return CustomRecognition.AnalyzeResult(
+                box=adjusted, detail={"result": "hit", "preset": preset_node})
+
+        # 真未命中：阶段原因已由预设节点(独立模式)记进嵌套识别记录；这里附带透传其 raw_detail
```

Ruff (0.15.15)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)